Revised MultiSelect.md (#2389)
coveo/plasma: packages/website/docs/MultiSelect.md

## Best Practices

Use a Multi select to have users **filter and select options from a long list**. A Multi selects is especially appropriate when the available list of options is very long or when the space is limited.

List options in alphanumerical order unless a more suited ordering rationale applies, for example when listing size or security level options.

Always include the ability to filter options when the list contains 20 or more.

## Labeling

Keep titles and labels short, preferably **under three words**.

### Title

The title should indicate the type of information to provide.

Provide a descriptive title without action verbs. For example, write "Grocery list" rather than "Select all grocery items".

### Labels

> Labels identify each option and should be self-explanatory. The width of the input should allow to fully display the name of the selected option.

Reviewer: ```suggestion
Labels identify each option and should be self-explanatory. The width of the input should allow to fully display the label of the selected option.
```

Use a consistent writing style for all options in the list.

## Help Text and Instructions

The placeholder text should indicate the type of information to select. Use an action verb. For example, write "Select an ingredient" rather than "Select an option".

## Feedback and Validation

Allow the addition of custom values only when it doesn't increase the risk of failure and when there may be options other than those listed.

Examples:

> -   When users select countries to filter on: do not allow custom values.

Reviewer: ```suggestion
-   When users select countries: do not allow custom values.
```

-   When users build a grocery list: allow custom values.

---

## Related Components

If your use case doesn't match the guidelines above, consider using one of the following components instead:

> -   [Multiline box](#/form/MultilineBox) if real estate is not an issue, or if legibility of the selected option is critical.

Reviewer: ```suggestion
-   [Multiline box](#/form/MultilineBox) if space is not an issue or if legibility of the selected option is critical.
```

Reviewer: Also, I realize I don't understand the 2nd part, because at line 21 we say that the selected option must appear fully.

[Description]: # 
[Multi-select dropdown allow to select multiple options from a predefined list. It is typically used when there is a large number of available options.]: # 

# Best Practices 
Use multi-select to allow user to select options from a **long list that can be filtered**. Multi-selects are more appropriate when the available list of options is very long or when the space is limited. 
 
Use alphanumerical order unless a stronger ordering rational applies (such as sizes or security levels). 
 
Always include the ability to filter the list when there are over 20 options. 
 
 
## Labeling

> Keep labels short, **under one line**. 

Author: Suggestion: Keep titles and labels short

The title is used to indicate the type of information to be provided. 
 
Use descriptive labels without action verbs. 
- E.g. Use "Grocery items" instead of "Select all items for your grocery". 
 
### Labels
Labels are used to name each option. The width of the input should allow to fully display the name of the chosen option. 
 
Use a consistent writing style for all options in the list. 
 
 
## Help text and instructions
The placeholder text should indicate the type of information that must be selected with an action verb.
- E.g. Use "Select a condiment" instead of "Select an option". 
 
 
## Feedback and validation
Allow the addition of custom values only when it doesn't increase the risk of failure and when the availability of all possible options in the list can not be ensured. 
 
Here some examples: 
- When selecting countries to filter on: do not allow custom values. 
- When filling a grocery list: allow custom values. 
 
---
 
## Related Components
If your use case doesn't match these guidelines, consider using an alternate component instead. 
 
 - [Multiline Box](#/input/MultilineBox) if real estate is not an issue, or if legibility of the selected option is critical. 

>  - [Checkbox](#/components/NumericInput) if you have less than 10 options. 

Author: Correction:
#/input/Checkbox

## Best Practices

> Use multi-select to allow user to select options from a **long list that can be filtered**. Multi-selects are more appropriate when the available list of options is very long or when the space is limited.

Reviewer: ```suggestion
Use a Multi select to allow users to **filter and select options from a long list**. A Multi selects is especially appropriate when the available list of options is very long or when the space is limited.
```

> Use alphanumerical order unless a more suited ordering rational applies (such as sizes or security levels).

Reviewer: ```suggestion
List options in alphanumerical order unless a more suited ordering rationale applies, for example when listing size or security level options.
```

> Always include the ability to filter the list when there are over 20 options.

Reviewer: ```suggestion
Always include the ability to filter options when the list contains 20 or more.
```

> ## Labeling

Reviewer: ```suggestion
## Text
```

Author: Will keep labeling

> Keep titles and labels short, **under one line**.

Reviewer: In the two other PRs I reviewed, we had:
```suggestion
Keep titles and labels short, preferably **under three words**.
```
I don't know if we want to say the same here? At the top of my head, I can't see how this is different from the other components?

### Title

> The title is used to indicate the type of information to be provided.

Reviewer: ```suggestion
The title should indicate the type of information to provide.
```

> Use descriptive labels without action verbs.
> 
> -   E.g. Use "Grocery items" instead of "Select all items for your grocery".

Reviewer: ```suggestion
Provide a descriptive title without action verbs. For example, write "Grocery items" rather than "Select all grocery items".
```

### Labels

> Labels are used to name each option. The width of the input should allow to fully display the name of the chosen option.

Reviewer: ```suggestion
Labels identify each option and should be self-explanatory. The width of the input should allow to fully display the name of the selected option.
```

Use a consistent writing style for all options in the list.

> ## Help text and instructions

Reviewer: ```suggestion
## Help Text and Instructions
```

> The placeholder text should indicate the type of information that must be selected with an action verb.
> 
> -   E.g. Use "Select a condiment" instead of "Select an option".

Reviewer: ```suggestion
The placeholder text should indicate the type of information to select. Use an action verb. For example, write "Select a condiment" rather than "Select an option".
```

> ## Feedback and validation

Reviewer: ```suggestion
## Feedback and Validation
```

> Allow the addition of custom values only when it doesn't increase the risk of failure and when the availability of all possible options in the list can not be ensured.

Reviewer: ```suggestion
Allow the addition of custom values only when it doesn't increase the risk of failure and when there may be options other than those listed.
```

Author: Same comment here applies vs Single Select

> Here some examples:
> 
> -   When selecting countries to filter on: do not allow custom values.
> -   When filling a grocery list: allow custom values.

Reviewer: ```suggestion
Examples:
-   When making users select countries to filter on: do not allow custom values.
-   When making users build a grocery list: allow custom values.
```

---

## Related Components

> If your use case doesn't match these guidelines, consider using an alternate component instead.

Reviewer: ```suggestion
If your use case doesn't match the guidelines above, consider using one of the following components instead:
```

> -   [Multiline Box](#/form/MultilineBox) if real estate is not an issue, or if legibility of the selected option is critical.

Reviewer: ```suggestion
-   [Multiline box](#/form/MultilineBox) if real estate is not an issue, or if legibility of the selected option is critical.
```
@kpoulincoveo Pas sûre de comprendre: à la ligne 23, on dit que le label au complet devrait apparaître?

Author: @jlevesquecoveo yes but they are much smaller and stacked closely together while the multiline box has more space which increase legibility.

> -   [Checkbox](#/form/Checkbox) if you have less than 10 options.

Reviewer: ```suggestion
-   [Checkbox](#/form/Checkbox) if there are seven options or less.
```

> -   [Single-Select](#/form/SingleSelect) if you can select only one option from a the list.

Reviewer: ```suggestion
-   [Single select](#/form/SingleSelect) if users can select only one option from the list.
```

[Description]: # 
[Multi-select dropdown allow to select multiple options from a predefined list. It is typically used when there is a large number of available options.]: # 

## Best Practices 
Use multi-select to allow user to select options from a **long list that can be filtered**.
More appropriate when the available list of options is very long or when real estate is limited.

> Use alphanumerical order unless a stronger ordering rational applies (such as numerical for prices, or custom for security levels).

Reviewer 2: suggestion : stronger -> more suited
